enterprise/com/src/test/java/org/neo4j/com/storecopy/FileMoveProviderTest.java

```diff
@@ -0,0 +1,258 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.com.storecopy;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.neo4j.graphdb.mockfs.EphemeralFileSystemAbstraction;
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.fs.StoreChannel;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.test.rule.PageCacheRule;
+import org.neo4j.test.rule.TestDirectory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileMoveProviderTest
+{
+    private DefaultFileSystemAbstraction defaultFileSystemAbstraction = new DefaultFileSystemAbstraction();
+    private EphemeralFileSystemAbstraction ephemeralFileSystemAbstraction = new EphemeralFileSystemAbstraction();
+
+    @Rule
+    public TestDirectory testDirectory = TestDirectory.testDirectory( defaultFileSystemAbstraction );
+
+    private FileMoveProvider subject;
+
+    public PageCacheRule pageCacheRule = new PageCacheRule();
+    private PageCache pageCache;
+    private FileMoveActionInformer fileMoveActionInformer;
+
+    @Before
+    public void setup()
+    {
+        pageCache = pageCacheRule.getPageCache( ephemeralFileSystemAbstraction );
+        fileMoveActionInformer = mock( FileMoveActionInformer.class );
+        subject = new FileMoveProvider( pageCache, fileMoveActionInformer );
+    }
+
+    @Test
+    public void moveSingleFiles()
+    {
+        // given
+        File sharedParent = testDirectory.directory( "shared_parent" );
+        File sourceParent = createDirectory( new File( sharedParent, "source" ) );
+        File sourceFile = createFile( new File( sourceParent, "file.txt" ) );
+        writeToFile( sourceFile, "Garbage data" );
+        File targetParent = createDirectory( new File( sharedParent, "target" ) );
+        File targetFile = new File( targetParent, "file.txt" );
+
+        // when
+        subject.traverseGenerateMoveActions( sourceFile, sourceParent ).forEach( moveToDirectory( targetParent ) );
+
+        // then
+        assertEquals( "Garbage data", readFromFile( targetFile ) );
+    }
+
+    private interface RunnableThrowable
+    {
+        void run() throws Throwable;
+    }
```

Comment:
This is the same as `org.neo4j.function.ThrowingAction`, so you can use that instead.